packages/puppeteer-core/src/common/util.ts

```diff
@@ -594,9 +593,7 @@ export async function waitForHTTP<T extends {url(): string}>(
   cancelation: Deferred<never>
 ): Promise<T> {
   return await firstValueFrom(
-    (
-      fromEvent(networkManager, eventName as unknown as string) as Observable<T>
-    ).pipe(
+    (fromEmitterEvent(networkManager, eventName) as Observable<T>).pipe(
```

Comment:
Can we use `eventName: NetworkManagerEvents`? and remove this type casting?

Reply:
Looks non-trivial.